src/sphericalHarmonicsExpansion.jl

```diff
@@ -1,10 +1,10 @@
 import Base.setindex!, Base.getindex, Base.isapprox, Base.==, Base.!=,
         Base.+, Base.-, Base.*, Base./, Base.write
 
-mutable struct SphericalHarmonicCoefficients{T<:Real}
-  c::AbstractVector{T}
-  L::Integer
-  R::Real
+mutable struct SphericalHarmonicCoefficients{T<:Real, vecT<:AbstractVector{T}}
+  c::vecT
+  L::Int64
+  R::Float64
```

Comment:
I would change the type of `R` to `T` instead, that way we can actually dispatch on the type `T`, otherwise it has no use in the parametric type in my opinion.
```suggestion
  R::T
```

Reply:
This would definitely be implementable given a few additional changes, but in this state, one runs into issues with potential type mismatches in arithmetic operations when the coefficients are passed as integers. This is supported by 2 tests erroring when implementing this change. This is why I went for the more pragmatic static type approach, since I'd assume it covers all relevant use cases and doesn't run into any trouble with arithmetic. We then still need the parametric T, since AbstractVector{Real} results in the aforementioned type instability.

Reply:
Ok that's true, I haven't thought about integer inputs. Maybe it's more sensible to switch to `T<:AbstractFloat`. However, we should have a look how our constructors handle integer vectors currently.

Reply:
That sounds like a reasonable idea. Currently, that would involve either rejecting integer coefficient inputs (which I don't think we want) or adding a conversion constructor to promote integer vectors to vectors of Float64.